bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -76,27 +70,22 @@ public AudioStream getInputStream(AudioFormat audioFormat) throws AudioException
                     if (clientSocketLocal == null) {
                         break;
                     }
-                    var sourceFormat = pulseaudioHandler.getSourceAudioFormat();
-                    if (sourceFormat == null) {
-                        throw new AudioException("Unable to get source audio format");
-                    }
-                    if (!audioFormat.isCompatible(sourceFormat)) {
+                    if (!audioFormat.isCompatible(streamFormat)) {
                         throw new AudioException("Incompatible audio format requested");
                     }
-                    var pipeOutput = new PipedOutputStream();
-                    var pipeInput = new PipedInputStream(pipeOutput, 1024 * 10) {
-                        @Override
-                        public void close() throws IOException {
-                            unregisterPipe(pipeOutput);
-                            super.close();
+                    var audioStream = streamGroup.getAudioStreamInGroup();
+                    audioStream.onClose(() -> {
+                        minusClientCount();
+                        try {
+                            Thread.sleep(0);
+                        } catch (InterruptedException ignored) {
```

Comment:
What is the purpose of this sleep with 0 delay ?

Reply:
It should be there before refactoring it, I have checked with the latest snapshot and it works correctly without it.